src/sonic-yang-models/yang-models/sonic-extension.yang

```diff
@@ -7,7 +7,54 @@ module sonic-extension {
 
     description "Extension yang Module for SONiC OS";
 
+    revision 2021-02-23 {
+        description "Adding extensions as per Sonic Yang Guidelines";
+    }
+
     revision 2019-07-01 {
         description "First Revision";
     }
+
+    /* For complete guide of using these extensions in SONiC yangs, refer
+    SONiC yang guidelines at
+    https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md */
+
+    extension db-name {
+        description "DB name, e.g. APPL_DB, CONFIG_DB";
+        argument "value";
+    }
+
+    extension key-delim {
```

Comment:
Few of the feature yang are in process of upstreaming that will be using non-config DB extensions. We have to be ready before they upstream to avoid build failures.

Reply:
We have libraries which gives us this delimiter for a DB. This seems redundant to me.

Reply:
When SONiC yangs are used as Northbound yang, mgmt-framework needs both db-name and key-delim extensions and these are used only for non CONFIG_DB fields.